### PR TITLE
Add support for `XP-Pen Deco M`

### PR DESCRIPTION
**Pull request by hax0r31337**
Idk what is `discord verification link` for, if it was required, please tell me how to get verified and I'll add it

### CODE REVIEW

**Comment by jamesbt365** [comment]
help > export diagnostics, save that file and upload here to this pull request for verification purposes, if you know for a fact that everything works and the correct values show in the tablet > tablet debugger, you will not need to confirm anything but that at this time.

**Comment by hax0r31337** [comment]
> help > export diagnostics, save that file and upload here to this pull request for verification purposes, if you know for a fact that everything works and the correct values show in the tablet > tablet debugger, you will not need to confirm anything but that at this time.

Since the diagnostics data includes some sensitive data which I don't want to publish on Github, may I contact you on Discord?

**Comment by jamesbt365** [comment]
The only "private" information that should be within the file is your PATH (environment variable), possibly other environment variables, and your username on your computer if you prefer to keep that private.

The main part of importance is the tablet devices itself in the HID devices section.

You may cut out any unrelated device.

**Comment by hax0r31337** [comment]
> The only "private" information that should be within the file is your PATH (environment variable), possibly other environment variables, and your username on your computer if you prefer to keep that private.
> 
> The main part of importance is the tablet devices itself in the HID devices section.
> 
> You may cut out any unrelated device.

That's ok, I'll do it later when I'm back from work

**Comment by hax0r31337** [comment]
Sorry for the delay!   

```json
{
  "AppVersion": "OpenTabletDriver v0.7.0.0+519e3c7d",
  "BuildDate": "",
  "OperatingSystem": {
    "Name": "Arch Linux",
    "Version": "Unknown",
    "Attributes": {
      "PRETTY_NAME": "Arch Linux",
      "ID": "arch",
      "BUILD_ID": "rolling",
      "KERNEL_VERSION": "6.6.1-arch1-1"
    }
  },
  "Devices": [
    {
      "VendorID": 10429,
      "ProductID": 2358,
      "InputReportLength": 8,
      "OutputReportLength": 0,
      "FeatureReportLength": 0,
      "Manufacturer": "Hanvon Ugee",
      "ProductName": "Deco M",
      "FriendlyName": "Deco M",
      "SerialNumber": "0000000000",
      "DevicePath": "/sys/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.0/0003:28BD:0936.0006/hidraw/hidraw1",
      "CanOpen": true
    },
    {
      "VendorID": 10429,
      "ProductID": 2358,
      "InputReportLength": 10,
      "OutputReportLength": 0,
      "FeatureReportLength": 0,
      "Manufacturer": "Hanvon Ugee",
      "ProductName": "Deco M",
      "FriendlyName": "Deco M",
      "SerialNumber": "0000000000",
      "DevicePath": "/sys/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.1/0003:28BD:0936.0007/hidraw/hidraw2",
      "CanOpen": true
    },
    {
      "VendorID": 10429,
      "ProductID": 2358,
      "InputReportLength": 12,
      "OutputReportLength": 33,
      "FeatureReportLength": 0,
      "Manufacturer": "Hanvon Ugee",
      "ProductName": "Deco M",
      "FriendlyName": "Deco M",
      "SerialNumber": "0000000000",
      "DevicePath": "/sys/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.2/0003:28BD:0936.0008/hidraw/hidraw4",
      "CanOpen": true
    }
  ],
  "ConsoleLog": [
    {
      "Time": "2023-11-15T19:44:03.2613421+08:00",
      "Group": "RootHub",
      "Message": "Initialized internal child hubs: HidSharpDeviceRootHub",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2615343+08:00",
      "Group": "Detect",
      "Message": "Searching for tablets...",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2615977+08:00",
      "Group": "Detect",
      "Message": "No tablets were detected.",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2616521+08:00",
      "Group": "IPC",
      "Message": "Connected to a client.",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2617022+08:00",
      "Group": "IPC",
      "Message": "Disconnected from a client.",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2617586+08:00",
      "Group": "DriverDaemon",
      "Message": "Sleep detected...",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2618052+08:00",
      "Group": "Detect",
      "Message": "Searching for tablets...",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2618872+08:00",
      "Group": "Detect",
      "Message": "No tablets were detected.",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.261944+08:00",
      "Group": "DriverDaemon",
      "Message": "Sleep detected...",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2619918+08:00",
      "Group": "Detect",
      "Message": "Searching for tablets...",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2620437+08:00",
      "Group": "Detect",
      "Message": "No tablets were detected.",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2620915+08:00",
      "Group": "HidSharpDeviceRootHub",
      "Message": "Changes: 1, Add: 0, Remove: 1",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2621437+08:00",
      "Group": "HidSharpDeviceRootHub",
      "Message": "Changes: 1, Add: 0, Remove: 1",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.262192+08:00",
      "Group": "RootHub",
      "Message": "Debounced HidSharpDeviceRootHub's DevicesChanged event",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2622412+08:00",
      "Group": "RootHub",
      "Message": "Invoking DevicesChanged",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2623078+08:00",
      "Group": "HidSharpDeviceRootHub",
      "Message": "Changes: 1, Add: 0, Remove: 1",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.262361+08:00",
      "Group": "RootHub",
      "Message": "Invoking DevicesChanged",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2624317+08:00",
      "Group": "HidSharpDeviceRootHub",
      "Message": "Changes: 1, Add: 1, Remove: 0",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2625004+08:00",
      "Group": "RootHub",
      "Message": "Invoking DevicesChanged",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2625864+08:00",
      "Group": "Detect",
      "Message": "Searching for tablets...",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2626645+08:00",
      "Group": "Detect",
      "Message": "No tablets were detected.",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2627436+08:00",
      "Group": "HidSharpDeviceRootHub",
      "Message": "Changes: 3, Add: 3, Remove: 0",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2628208+08:00",
      "Group": "RootHub",
      "Message": "Invoking DevicesChanged",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2628958+08:00",
      "Group": "Detect",
      "Message": "Searching for tablets...",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2629743+08:00",
      "Group": "Detect",
      "Message": "Attempting to match config 'XP-Pen Deco M' to device 'Deco M (/sys/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.0/0003:28BD:0936.0006/hidraw/hidraw1)'",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2630503+08:00",
      "Group": "Detect",
      "Message": "Attempting to match config 'XP-Pen Deco M' to device 'Deco M (/sys/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.1/0003:28BD:0936.0007/hidraw/hidraw2)'",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2631383+08:00",
      "Group": "Detect",
      "Message": "Attempting to match config 'XP-Pen Deco M' to device 'Deco M (/sys/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.2/0003:28BD:0936.0008/hidraw/hidraw4)'",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2632189+08:00",
      "Group": "Detect",
      "Message": "Found 'XP-Pen Deco M' digitizer: 'Deco M (/sys/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.2/0003:28BD:0936.0008/hidraw/hidraw4)'",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2633006+08:00",
      "Group": "Detect",
      "Message": "Found tablet 'XP-Pen Deco M'",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2633868+08:00",
      "Group": "Device",
      "Message": "Initializing device 'Deco M (/sys/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.2/0003:28BD:0936.0008/hidraw/hidraw4)'",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2635272+08:00",
      "Group": "Device",
      "Message": "Using report parser type 'OpenTabletDriver.Configurations.Parsers.XP_Pen.XP_PenOffsetPressureReportParser'",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2636125+08:00",
      "Group": "Device",
      "Message": "Set device output: 02-B0-04",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2636902+08:00",
      "Group": "Evdev",
      "Message": "Successfully initialized virtual pressure sensitive tablet. (code NONE)",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2637716+08:00",
      "Group": "LinuxArtistMode",
      "Message": "Input: [204x128@<102, 64>:0°]",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2638535+08:00",
      "Group": "LinuxArtistMode",
      "Message": "Output: [1920x1080@<960, 540>]",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2639348+08:00",
      "Group": "LinuxArtistMode",
      "Message": "LockAspectRatio: False",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2640162+08:00",
      "Group": "LinuxArtistMode",
      "Message": "AreaClipping: True",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2640917+08:00",
      "Group": "LinuxArtistMode",
      "Message": "AreaLimiting: True",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2641704+08:00",
      "Group": "LinuxArtistMode",
      "Message": "LockToBounds: True",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2642523+08:00",
      "Group": "XP-Pen Deco M",
      "Message": "Output mode: Artist Mode",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2643322+08:00",
      "Group": "MouseBinding",
      "Message": "Button: Left",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2644131+08:00",
      "Group": "MouseBinding",
      "Message": "Button: Right",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2644949+08:00",
      "Group": "Settings",
      "Message": "Driver is enabled.",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.264568+08:00",
      "Group": "LinuxArtistMode",
      "Message": "Input: [204x128@<102, 64>:0°]",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2646396+08:00",
      "Group": "LinuxArtistMode",
      "Message": "Output: [1920x1080@<960, 540>]",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2647169+08:00",
      "Group": "LinuxArtistMode",
      "Message": "LockAspectRatio: False",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2648001+08:00",
      "Group": "LinuxArtistMode",
      "Message": "AreaClipping: True",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2648871+08:00",
      "Group": "LinuxArtistMode",
      "Message": "AreaLimiting: True",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2649694+08:00",
      "Group": "LinuxArtistMode",
      "Message": "LockToBounds: True",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2650488+08:00",
      "Group": "XP-Pen Deco M",
      "Message": "Output mode: Artist Mode",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2651305+08:00",
      "Group": "MouseBinding",
      "Message": "Button: Left",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2652877+08:00",
      "Group": "MouseBinding",
      "Message": "Button: Right",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2653788+08:00",
      "Group": "Settings",
      "Message": "Driver is enabled.",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2023-11-15T19:44:03.2654575+08:00",
      "Group": "IPC",
      "Message": "Connected to a client.",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    }
  ]
}
```

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Deco M.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "Name": "XP-Pen Deco M",
+  "Specifications": {
+    "Digitizer": {
+      "Width": 203.2,
+      "Height": 127,
+      "MaxX": 40640,
+      "MaxY": 25400
+    },
+    "Pen": {
+      "MaxPressure": 8192,
```
Comment (gonX) [inline OpenTabletDriver.Configurations/Configurations/XP-Pen/Deco M.json:11]: This is usually 8191, though I see the Deco L also uses 8192. Please double check with the Tablet Debugger inside of OpenTabletDriver whether 8192 is truly max achievable pressure.

Comment (hax0r31337) [reply OpenTabletDriver.Configurations/Configurations/XP-Pen/Deco M.json]: Sorry for my oversight, the max pressure was actually `8191`, I based the config on the Deco L config without double checking this field.

**Review by gonX — APPROVED** [review]

**Pull request merged**